tests/unit/test_charm_status.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import unittest
+from unittest.mock import patch
+
+import hypothesis.strategies as st
+from helpers import patch_network_get
+from hypothesis import assume, given
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+from charm import PrometheusCharm
+
+logger = logging.getLogger(__name__)
+
+
+class TestActiveStatus(unittest.TestCase):
+    """Feature: Charm's status should reflect the correctness of the config / relations.
+
+    Background: When launched on its own, the charm should always end up with active status.
+    In some cases (e.g. Ingress conflicts) the charm should go into blocked state.
+    """
+
+    def setUp(self) -> None:
+        self.app_name = "prometheus-k8s"
+
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
+    @given(st.booleans(), st.integers(1, 5))
+    def test_unit_is_active_when_deployed_in_isolation(self, is_leader, num_units):
+        """Scenario: Unit is deployed without any user-provided config or regular relations.
+
+        The charm should end up in ActiveState when deployed in isolation with defaults only.
```

Comment:
1. The purpose of this test is not clear to me. The method name suggests this test is checking that Prometheus is active when it is "deployed_in_isolation". The docstring suggests that test is checking if Prometheus is active "without any user-provided config or regular relations". The body of the test varies the number of deployed peer units from 1 through 5. Shouldn't the number of units be 1 if we are trying to "deploy in isolation" ?
2. It is not clear to me if and what "deploying in isolation" has anything to do with Prometheus reload behaviour .
3. If there is something new here, could [this existing test](https://github.com/canonical/prometheus-k8s-operator/blob/main/tests/unit/test_charm.py#L184) be modified to fulfil that requirement ?